guava/src/main/kotlin/com/spectralogic/kotlinutils/guava/Memoize.kt

```diff
@@ -0,0 +1,41 @@
+package com.spectralogic.kotlinutils.guava
+
+import com.google.common.cache.Cache
+import com.google.common.cache.CacheBuilder
+
+fun <A> (() -> A).memoize(memoSize: Long = 1000L): () -> A {
```

Comment:
Add Javadoc to explain how these should be used.  Provide some usage examples as well.